File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/circuitbreakerstats/ServiceWindow.java
package uk.nhs.nhsx.circuitbreakerstats;

import java.time.*;

class ServiceWindow {
    private final LocalDate yesterday;
    private final LocalDate today;


    ServiceWindow(Instant time) {
        today = time.atZone(ZoneOffset.UTC).toLocalDate();
        yesterday = today.minusDays(1);
    }

    long queryStart() {
        return ZonedDateTime.of(yesterday, LocalTime.of(0, 0, 0), ZoneOffset.UTC).toEpochSecond();

    }

    long queryEnd() {
        return ZonedDateTime.of(yesterday, LocalTime.of(23, 59, 59), ZoneOffset.UTC).toEpochSecond();

    }
}

File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/analyticslogs/ServiceWindow.kt
package uk.nhs.nhsx.analyticslogs

import java.time.Instant
import java.time.LocalDate
import java.time.ZoneOffset.UTC

class ServiceWindow(time: Instant) {
    private val yesterday: LocalDate = LocalDate.ofInstant(time, UTC).minusDays(1)
    fun queryStart() = yesterday.atStartOfDay().toEpochSecond(UTC)
    fun queryEnd() = yesterday.atTime(23, 59, 59, 0).toEpochSecond(UTC)
}
